docs/api-reference/data-fetching/getServerSideProps.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Fetch data on each request with `getServerSideProps`. Learn more about this API for data fetching in Next.js.
+---
+
+# `getServerSideProps`
+
+<details>
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes                                                             |
+| --------- | ------------------------------------------------------------------- |
+| `v10.0.0` | `locale`, `locales`, `defaultLocale`, and `notFound` options added. |
+| `v9.3.0`  | `getServerSideProps` introduced.                                    |
+
+</details>
+
+When exporting an `async` function called `getServerSideProps` (server-side rendering) from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
+
+```js
+export async function getServerSideProps(context) {
+  return {
+    props: {}, // will be passed to the page component as props
+  }
+}
+```
+
+You can import modules in top-level scope for use in `getServerSideProps`. Imports used will **not be bundled for the client-side**. This means you can write **server-side code directly in `getServerSideProps`**, including reading from the filesystem or a database.
+
+You should not use the [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) `API` to call an [API route](/docs/api-routes/introduction.md) in `getServerSideProps`.
+
+Instead, directly import the logic used inside your `API` route. You may need to slightly refactor your code for this approach.
+
+> **Note:** The `fetch()` `API` _can_ be used to fetch external data, such as from a Content Management System (CMS) or `API`.
```

Comment:
IMO by removing the quote here, this lost some of the context of being a footnote rather than a main part of the document. To fix that, we need to likely pull this into a new section with more explanation (this is one of the most common confusion points around this).

You _can_ use fetch but it leads to a poor experience. You are adding an additional network hop, when instead, you can call that logic directly.

Reply:
Cool, can totally refactor this part. But I'm not sure I 100% understand your second paragraph. If I'm calling a CMS to get data, that I want rendered on the server, then I will use fetch to call the CMS API. This is the part I don't get: "you can call that logic directly"

Reply:
You'll occasionally see folks put this logic inside an API Route (which also runs on the server) and then call the API Route from gSP or gSSP. This _works_ but it's not recommended, since gSP and gSSP also run on the server. So rather than having an "additional hop" to the API route, you can just use the logic inside your API route directly!

Reply:
Ahh, gotcha! Thanks for the explanation!